### ISSUE TITLE
Profile dump not working when circuit is under high load

### ISSUE THREAD

**Issue by gz** [feldera/feldera#2063]
When the circuit is under high load, getting a profile dump fails with an HTTP timeout.

**Comment by blp** [comment]
If you use the direct URL to the pipeline, then it doesn't time out. I
think the pipeline manager should just be more patient with the timeout.

**Comment by lalithsuresh** [comment]
How did you reproduce this?

**Comment by blp** [comment]
> @blp How did you reproduce this?

I've been doing a lot of profiles on slow pipelines for a week or two. I usually had to run it multiple times until it didn't time out. I think that it only succeeds if the pipeline takes a step before the timeout duration expires.

**Comment by ryzhyk** [comment]
Yes, that's what happens. The way it used to work is that the API endpoint would simply initiate the profile dump, which actually happens at the end of the current step. Then the dump ends up in the file system, which was really inconvenient. @blp  converted it to a synchronous API call that returns the profile, but now we have the timeout issue.

Of course, the proper solution is to re-implement the internals of dump_profile so that it doesn't need to wait for `step` to complete. This should be possible.

**Comment by lalithsuresh** [comment]
I was going to suggest increasing the timeout for the call, but I guess the issue is that that doesn't help when the step doesn't complete.

**Comment by ryzhyk** [comment]
If the step doesn't complete at all then yes, it doesn't help. But even if it completes but takes a very long time, this is clearly not the best UX

**Comment by ryzhyk** [comment]
One workaround that works today, although clearly not ideal, is to pause the pipeline use `/profile_dump` in the paused state.

**Comment by mihaibudiu** [comment]
I assume that pausing won't work either under high load

**Comment by ryzhyk** [comment]
It should work, but potentially take a long time
